dev_docs/tutorials/adding_performance_metrics.mdx

```diff
@@ -0,0 +1,68 @@
+---
+id: kibDevTutorialAddingPerformanceMetrics
+slug: /kibana-dev-docs/tutorial/adding_performance_metrics
+title: Adding Performance Metrics
+summary: Learn how to instrument your code and analyze performance
+date: 2022-07-07
+tags: ['kibana', 'onboarding', 'setup', 'performance', 'development', 'telemetry']
+---
+
+Reporting performance events are easy as it can be seen below, however it's highly suggested to follow guidelines for what to report in certain fields:
+
+```typescript
+import { reportPerformanceMetricEvent } from '@kbn/ebt-tools';
+
+reportPerformanceMetricEvent(analytics, {
+  eventName: KIBANA_LOADED_EVENT,
+  meta: {
+    kibana_version: this.coreContext.env.packageInfo.version,
+    protocol      : window.location.protocol,
+    ...fetchOptionalMemoryInfo(),
+  },
+  duration: timing[LOAD_FIRST_NAV],
+  key1    : LOAD_START,
+  value1  : timing[LOAD_START],
+  key2    : LOAD_BOOTSTRAP_START,
+  value2  : timing[LOAD_BOOTSTRAP_START],
+  key3    : LOAD_CORE_CREATED,
+  value3  : timing[LOAD_CORE_CREATED],
+  key4    : LOAD_SETUP_DONE,
+  value4  : timing[LOAD_SETUP_DONE],
+  key5    : LOAD_START_DONE,
+  value5  : timing[LOAD_START_DONE],
+});
+```
+
+Normally reporting an event requires a registration of event type however since we want
+performance metrics to be uniform across all kibana we already summarized it with the structure above.
+We welcome all the feedback based on your usecases.
+
+# Guidelines of What to report in Performance Metric Events
+
+
+In order to simplify the consumption of performance telemetry, we'd like to standardize the shape of events. This requires considering the following aspects:
+- `Performance` - amount and size of events being sent
+- `Discoverability` - how easy it is to find different events, dimensions and metrics
+- `Queriability` - we would like to have the ability to write complex queries on top of telemetry data; for example getting all dashboard load events that took over 10 seconds for dashboards that contain less than 5 panels.
```

Comment:
Since we are using the dashboard example, do you think it's worth using the dashboard event in the call to `reportPerformanceMetricEvent`? This way, folks can look at where the number of panels is provided.

Reply:
I can add as an extra example 😇  since we already have kibana-loaded and there is some information relying on that example too.